src/frontend/js/components/user.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { PropTypes, Component } from 'react';
+
+export default class User extends Component {
+  static propTypes: {
+    id: PropTypes.string.isRequired,
+    user: PropTypes.object.isRequired,
+    fetchUser: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
```

Comment:
this is one of the scariest things available in ES6. I recommend you do your best not to use any inheritance features not imposed by the framework.

I also see you use constructor to do the initial fetch. It's not going to work if the component stays alive and then we need to reload its content.
If you're using `A extends Component` syntax from react, you have access to 
componentWillMount and componentWillReceiveProps that you can use to check if you should reload content and then do so.

This is the only thing I would try to block if I were doing a review of this pull request in real-life situation.

Is it something you found as a best practice? If so, I'll happily learn where I'm wrong about component lifecycle.